### PR TITLE
Shereen - Unit tests for SaveButton component.

### PR DESCRIPTION
**Pull request by pshereen**
# Description
Unit tests for Save Button component.

## How to test:
1. check into current branch
2. do `npm install` and `npm test SaveButton.test.js` to run this PR locally
3. Check all 5 tests are passed.

## Screenshots or videos of changes:

image

### CODE REVIEW

**Review by meetpadhiar4 — APPROVED** [review]
Hi, I tested your PR and it passed all the test cases. Please refer the screenshot attached.

PR 1977

**Review by vikrambadhan — APPROVED** [review]
Hi, I tested your PR and all the test cases passed!

Week 26 - PR2

**Review by ShengjieMao — APPROVED** [review]
All tests passed. Detailed and readable annotation in files. Can see comprehensive tests on components and props.

**Review by ChengyanOo — APPROVED** [review]
All tests passed!
Screenshot 2024-02-23 at 10 35 09

**Review by XiaohanMeng — APPROVED** [review]
Hi, all test cases passed. Well done!
#1977-result

File: src/components/UserProfile/UserProfileEdit/__test__/SaveButton.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        setSaved:jest.fn(),
+        } 
+      render(<SaveButton {...props} />);
+      const button = screen.getByRole('button', { name: /save changes/i });
+      expect(button).toBeInTheDocument(); 
+      expect(button).toBeDisabled(); 
+    });
+
+    // Test Case 2: It should render the modal after clicking save changes button
+    it('renders modal after save changes button is clicked', async () => {
```
Comment (FyisFe) [inline src/components/UserProfile/UserProfileEdit/__test__/SaveButton.test.js:37]: The `async` seems not necessary here

**Review by FyisFe — APPROVED** [review]
All tests passes, lgtm!

image

**Comment by beblicarl** [comment]
All tests passed successfully and the test cases are great

image

**Review by beblicarl — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**